platforms/core-configuration/model-core/src/test/groovy/org/gradle/api/internal/provider/CompoundAssignmentPropertySpec.groovy

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.provider
+
+import org.gradle.api.internal.provider.support.LazyGroovySupport
+import org.gradle.api.internal.provider.support.SupportsCompoundAssignment
+import org.gradle.api.provider.Provider
+import org.gradle.internal.evaluation.CircularEvaluationException
+import spock.lang.Specification
+
+abstract class CompoundAssignmentPropertySpec<T extends LazyGroovySupport & Provider<?>> extends Specification {
```

Comment:
🤔 While it's good to have additional test coverage, this test setup feel fragile. The content of each test mimics the behavior of an AST transformation. It means that if any change was to be made to that transformation, each test in this class would have to be manually updated. However, if we forget to update these tests, would not fail, because they don't depend on the transformation directly.

In other words, it looks more like a unit test for the logic of wrap/unwrap and its overloads. If that's the case, maybe it should be restructured?

I am definitely not against keeping the test as is, but wanted to share my concerns.

Reply:
Nice!